internal/collector/collector.go

```diff
@@ -113,8 +115,19 @@ func (k *KeepalivedCollector) Collect(ch chan<- prometheus.Metric) {
 
 	keepalivedUp := float64(1)
 
-	keepalivedStats, err := k.getKeepalivedStats()
-	if err != nil {
+	b := backoff.NewExponentialBackOff()
+	b.InitialInterval = 10 * time.Millisecond
+	b.MaxElapsedTime = 2 * time.Second
+	b.Reset()
+
+	var keepalivedStats *KeepalivedStats
+
+	if err := backoff.Retry(func() error {
+		var err error
+		keepalivedStats, err = k.getKeepalivedStats()
+
```

Comment:
To not make the logs look wired do you think adding a warning log that it will be retried in case the err is not nil in this retry func would be a good idea? As we are already logging e.g. `Failed to...` and this would repeat many times due to retry...